uefi/src/proto/device_path/mod.rs

```diff
@@ -159,6 +153,20 @@ impl<'a> TryFrom<&'a [u8]> for &'a DevicePathHeader {
     }
 }
 
+impl Deref for DevicePathHeader {
+    type Target = DevicePathProtocol;
```

Comment:
I'd like to avoid using `uefi-raw` types in the public API of `uefi`, except where we `pub use` those types. (Otherwise, code that uses `uefi` can't directly refer to those types unless it also adds a dep on `uefi-raw`, which shouldn't normally be necessary.)

In this case, could we remove the deref and instead provide three getter methods -- device_type, sub_type, and length? It's a little bit more code but I think a nicer API.

Reply:
Good catch, thanks! Didn't think about that.

Reply:
What do you think about the new version?

Reply:
Left a few minor comments and otherwise lgtm.